integrations/access/servicenow/client.go

```diff
@@ -141,15 +143,47 @@ func (conf ClientConfig) checkAndSetDefaults() error {
 	}
 	return nil
 }
+func onAfterServicenowResponse(sink common.StatusSink) resty.ResponseMiddleware {
+	return func(_ *resty.Client, resp *resty.Response) error {
+		if sink != nil {
+			var code types.PluginStatusCode
+			switch {
+			case resp.StatusCode() == http.StatusUnauthorized:
+				code = types.PluginStatusCode_UNAUTHORIZED
+			case resp.StatusCode() >= 200 && resp.StatusCode() < 400:
+				code = types.PluginStatusCode_RUNNING
+			default:
+				code = types.PluginStatusCode_OTHER_ERROR
+			}
+			if err := sink.Emit(resp.Request.Context(), &types.PluginStatusV1{Code: code}); err != nil {
+				log := logger.Get(resp.Request.Context())
+				log.ErrorContext(resp.Request.Context(), "Error while emitting servicenow plugin status",
+					"error", err,
+					"code", resp.StatusCode(),
+				)
+			}
+		}
+		if resp.IsError() {
+			return errWrapper(resp.StatusCode(), resp.Body())
+		}
+		return nil
+	}
+}
```

Comment:
This function is literally the same as in the Opsgenie package, except for maybe plugin name in the error message. Can we have a common "on after response" function and use for both plugins instead of duplicating?

Reply:
Have added in a generic OnAfterResponse to common, had to add in a common/helpers.go as I could not find a more appropriate place for it, I could move it to common/status.go but im not sure that is better.